Runtime/JSObject.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+
+namespace NodeApi;
+
+public struct JSObject : IEnumerable<(JSValue name, JSValue value)>, IEnumerable
+{
+    private JSValue _value;
+
+    public static explicit operator JSObject(JSValue value) => new() { _value = value };
+    public static implicit operator JSValue(JSObject obj) => obj._value;
+
+    public JSObject()
+    {
+        _value = JSValue.CreateObject();
+    }
+
+    public JSValue this[JSValue name]
+    {
+        get => _value.GetProperty(name);
+        set => _value.SetProperty(name, value);
+    }
+
+    public JSValue this[string name]
+    {
+        get => _value.GetProperty(name);
+        set => _value.SetProperty(name, value);
+    }
+
+    public void Add(JSValue name, JSValue value)
+        => _value.SetProperty(name, value);
+
+    public void Add((JSValue Key, JSValue Value) entry)
+        => _value.SetProperty(entry.Key, entry.Value);
+
+    public JSObjectPropertyEnumerator GetEnumerator() => new(_value);
```

Comment:
Similar to `JSArrayEnumerator`, I think `JSObjectEnumerator` can be internal, or private nested.

Reply:
Please see my other comment: I need it to be public to avoid memory allocation as it is done in `List.Enumerator`.

Reply:
I'd suggest nested `JSObject.Enumerator` then.